File: public/lib/fidl/dart/src/bindings/enum.dart
// Copyright 2016 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

part of bindings;

/// An enumerated typed in a FIDL interface.
class FidlEnum {
  /// The numerical value assigned to this value in the interface.
  final int fidlEnumValue;

  /// Creates an instance with the given numerical value.
  const FidlEnum(this.fidlEnumValue);
}

File: garnet/public/lib/fidl/tools/difl/enum.py
# Copyright 2019 The Fuchsia Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.


from typing import List, Optional, Sequence, Tuple, FrozenSet, Set, Dict, NamedTuple

from difl.ir import Library, Struct, StructMember, Type, Table, Union, Protocol, Enum
from difl.changes import *


def enum_changes(before: Enum, after: Enum, identifier_compatibility: Dict[str, bool]) -> List[Change]:
    changes: List[Change] = []

    # TODO: actually compare enums

    identifier_compatibility[before.name] = True

    return changes
